src/applications/resources-and-support/hooks/useGetSearchResults.js

```diff
@@ -65,18 +65,45 @@ export default function useGetSearchResults(articles, query, page) {
         ),
 
         // Number of times a keyword is found in the article's introText.
-        keywordsCountsDescription: keywords?.reduce(
+        keywordsCountsIntroText: keywords?.reduce(
           (keywordInstances, keyword) =>
             keywordInstances +
             article.introText.toLowerCase()?.split(keyword)?.length -
             1,
           0,
         ),
 
-        wholePhraseMatchCounts:
+        // Number of times a keyword is found in the article's searchableContent.
+        keywordsCountsContent: keywords?.reduce(
+          (keywordInstances, keyword) =>
+            keywordInstances +
+            article.searchableContent.toLowerCase()?.split(keyword)?.length -
+            1,
+          0,
+        ),
+
+        // Number of times the full query is found in the article's title.
+        wholePhraseMatchCountsTitle:
+          article.title.toLowerCase()?.split(query.toLowerCase())?.length - 1,
+
+        // Number of times the full query is found in the article's introText.
+        wholePhraseMatchCountsIntroText:
+          article.introText.toLowerCase()?.split(query.toLowerCase())?.length -
+          1,
+
+        // Number of times the full query is found in the article's searchableContent.
+        wholePhraseMatchCountsContent:
+          article.searchableContent.toLowerCase()?.split(query.toLowerCase())
+            ?.length - 1,
+
+        // Number of times the full query is found in the article's title, introText, searchableContent combined.
+        wholePhraseMatchCountsTotal:
           article.title.toLowerCase()?.split(query.toLowerCase())?.length -
           1 +
           (article.introText.toLowerCase()?.split(query.toLowerCase())?.length -
+            1) +
+          (article.searchableContent.toLowerCase()?.split(query.toLowerCase())
+            ?.length -
```

Comment:
I'm wondering if we might be able to refactor to abstract some of this into a helper or store them in variables?

It looks like `wholePhraseMatchCountsTotal` is the sum of `wholePhraseMatchCountsTitle` + `wholePhraseMatchCountsIntroText` + `wholePhraseMatchCountsContent`, unless I'm reading it wrong.

It also seems like this line `article.SOME_PROPERTY.toLowerCase()?.split(query.toLowerCase())?.length - 1` could be a nice helper function. My coffee is still kicking but right now that line is hard for me to understand.

Reply:
Updated